iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/NamingStrategy.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.modification.topology;
+
+import com.powsybl.iidm.network.*;
+
+import java.util.List;
+
+/**
+ * The naming strategy aims at clarifying and facilitating the naming of the different network elements created via the
+ * different {@link com.powsybl.iidm.modification.NetworkModification} classes. Based on the name of the network element
+ * the user wish to create (a VoltageLevel, a BranchFeederBay, etc.), all the other elements created during the
+ * NetworkModification will be given a name using this name as baseline and prefixes/suffixes according to the naming
+ * strategy chosen by the user. The naming strategy can be either the default one
+ * ({@link com.powsybl.iidm.modification.topology.DefaultNamingStrategy}) or a new implementation of the present class
+ * provided by the user.
+ *
+ * @author Nicolas Rol <nicolas.rol@rte-france.com>
+ */
+public interface NamingStrategy {
+
+    String getName();
+
+    String getSectioningPrefix(String baseId, BusbarSection bbs, int busBarNum, int section1Num, int section2Num);
+
+    String getChunkPrefix(String baseId, List<SwitchKind> switchKindList, int busBarNum, int section1Num, int section2Num);
+
+    String getDisconnectorId(String baseId, int id1Num, int id2Num);
+
```

Comment:
I think it is too many method with the same name but different arguments. It makes it difficult to implement. In the interface, there should be as little methods as possible but then the implementation can be more complex. Maybe using an enumeration indicating what is the purpose of the name could be useful?
Also it's a bit confusing that there is a getSwitchId and getDisconnectorId/getBreakerId. This could be simplified with topology kind and switch type in the arguments.

Reply:
The problem I see is that, if we choose to have only one method for every use of switches, we'll have a method with many parameters and we'll have to use a lot of `null` when using it in the network modifications.
For the type of switches, it might be possible to do it but we'll also have to use some null when calling methods in some cases

Reply:
I will take sides with "the fewer arguments, the better" but I get the fact that having a complex interface is not very user-friendly...